lib/nostr/nostr_api.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:convert';
+
+import 'package:cake_wallet/generated/i18n.dart';
+import 'package:cake_wallet/nostr/nostr_user.dart';
+import 'package:cake_wallet/src/widgets/alert_with_one_action.dart';
+import 'package:cake_wallet/src/widgets/picker.dart';
+import 'package:cake_wallet/utils/show_pop_up.dart';
+import 'package:flutter/material.dart';
+import 'package:nostr_tools/nostr_tools.dart';
+
+class NostrProfileHandler {
+  static final relayToDomainMap = {
+    'relay.snort.social': 'snort.social',
+  };
+
+  static Nip05 nip05 = Nip05();
+  static final noRelaysTitle = 'No relays';
+  static final pickRelayTitle = 'Please choose a relay to use';
+  static final noRelaysMessage =
+      'We found a Nostr NIP-05 record for this user, but it does not contain any relays. Please instruct the recipient to add relays to their Nostr record.';
+  static final noUsersDomainRelay =
+      'There isn\'t a relay for user\'s domain or the relay is unavailable. Please choose a relay to use.';
```

Comment:
localization is missing.
and since you are already passing the `context` then you can use `S.of(context)` in the functions instead of having static variables with predefined values